### ISSUE TITLE
Add head_mask for transformers

### ISSUE THREAD

**Issue by rowedenny** [pytorch/pytorch#114796]
### 🚀 The feature, motivation and pitch

I am wondering if we can also have the head_mask, which can manually alter the attention_weights after the softmax
[From Huggingface](https://github.com/huggingface/transformers/blob/main/src/transformers/models/roberta/modeling_roberta.py#L272C1-L274C58)

**Comment by mikaylagawarecki** [comment]
cc @drisspg on whether this can be done from `attn_mask`
